Library/src/Systems/Settings/SettingsSystem.cpp

```diff
@@ -943,10 +926,9 @@ void SettingsSystem::GetAvatarInfo(AvatarInfoResultCallback Callback)
 			return;
 		}
 
-		const auto& type		   = Json["type"];
-		const auto& identifierType = Json["identifierType"];
+		const auto& type = Json["type"];
 
-		if (!type.IsInt() || !identifierType.IsInt())
```

Comment:
[Consider] I wonder if we should keep the identifierType check here, ie, assert that it's a string. I know it must always be one from now on, but I'm wondering about old data, (presuming that these json documents are persisted somewhere). Even with that aside, feels like decent practice to guarentee the types of data coming out of untyped json.

Reply:
In ed4038b65734569d0604bc55e697ac9f69707415, I have re-instated the identifierType field in the json, hard-coded it to string in all new json files, and added a check to make sure it is a string when loading json files. This will now generate an error for older json files that still contain an int type. (if they exist)

Reply:
Following our discussion with @MAG-mv, I have made it silently convert any old-style integers to string. I have also added a test case that ensures that ints are treated gracefully, should any of them by lying around. I have removed identifierType, we no longer set it.